### PR TITLE
TINY-6646: Fixed incorrectly calculating col widths when resizing tables

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-6646

Description of Changes:
This fixes the widths being calculated incorrectly for `col` elements without raw widths. The problem was most browsers will return 0px for the width, as such if there's no raw width we need to fallback to getting the width from the table cells. It also fixes an issue I found with the fallback logic, as it would never calculate the last column width correctly.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)
* [x] ~License headers added on new files (if applicable)~

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/changelog.txt [hunk 1]
```diff
@@ -30,6 +30,7 @@ Version 5.6.0 (TBD)
     Fixed an issue where menus would incorrectly collapse in small containers #TINY-3321
     Fixed some minor memory leaks that prevented garbage collection for editor instances #TINY-6570
     Fixed resizing a `responsive` table not working when using the column resize handles #TINY-6601
+    Fixed incorrectly calculating table `col` widths resizing responsive tables #TINY-6646
```
Comment (ltrouton) [inline modules/tinymce/changelog.txt:33]:
```suggestion
    Fixed incorrectly calculating table `col` widths when resizing responsive tables #TINY-6646
```

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/changelog.txt (hunk 1 above)
Comment (lnewson) [reply modules/tinymce/changelog.txt]: Oops, thanks for catching that!

**Review by JamesToohey — APPROVED** [review]

File: modules/snooker/src/main/ts/ephox/snooker/resize/ColumnSizes.ts [hunk 2]
```diff
@@ -27,36 +31,52 @@ const getRawH = function (cell: SugarElement) {
...
-      return fallback(deduced);
-    }, function (cell) {
-      return getWidth(cell, tableSize);
-    });
+    return getDimension(cellOption, c, backups, colFilter, (column) => {
+      if (isValidColumn(column)) {
+        return getWidth(column, tableSize);
+      } else {
+        // Invalid column so fallback to trying to get the computed width from the cell
+        const cell = Optionals.bindFrom(columnCells[c], Fun.identity);
```
Comment (ashfordneil) [inline modules/snooker/src/main/ts/ephox/snooker/resize/ColumnSizes.ts:63]: Should this just be `Optional.from`?

**Review by ashfordneil — APPROVED** [review]

**Review by metricjs — APPROVED** [review]

File: modules/snooker/src/main/ts/ephox/snooker/resize/ColumnSizes.ts (hunk 2 above)
Comment (lnewson) [reply modules/snooker/src/main/ts/ephox/snooker/resize/ColumnSizes.ts:63]: No, that'd make the type `Optional<Optional<SugarElement>>` when we want it to be `Optional<SugarElement>`

**Pull request merged**